src/js/BookReader.js

```diff
@@ -940,58 +959,37 @@ BookReader.prototype.drawLeafsThumbnail = function(seekIndex) {
 };
 
 BookReader.prototype.lazyLoadThumbnails = function() {
-  // We check the complete property since load may not be fired if loading from the cache
-  this.$('.BRlazyloading').filter('[complete=true]').removeClass('BRlazyloading');
-
-  var loading = this.$('.BRlazyloading').length;
-  var toLoad = this.thumbMaxLoading - loading;
+  const self = this;
 
-  var self = this;
+  const batchImageRequestByRow = ($images) => {
+    utils.wait().then(() => {
+      $images.each(function loadEachImg() {
+        const $imgPlaceholder = this;
+        self.lazyLoadImage($imgPlaceholder)
+      });
+    })
+  };
 
-  if (toLoad > 0) {
-    // $$$ TODO load those near top (but not beyond) page view first
-    this.refs.$brPageViewEl.find('img.BRlazyload').filter(':lt(' + toLoad + ')').each( function() {
-      self.lazyLoadImage(this);
-    });
-  }
+  this.displayedRows.forEach((row) => {
+    const $imagesInRow = this.refs.$brPageViewEl.find(`[data-row="${row}"]`);
+    batchImageRequestByRow($imagesInRow);
+  });
 };
 
-BookReader.prototype.lazyLoadImage = function (dummyImage) {
-  var img = new Image();
-  var self = this;
-
-  $(img)
-    .addClass('BRlazyloading')
-    .one('load', function() {
-      $(this).removeClass('BRlazyloading').attr('alt', 'Loading book image');
-
-      // $$$ Calling lazyLoadThumbnails here was causing stack overflow on IE so
-      //     we call the function after a slight delay.  Also the img.complete property
-      //     is not yet set in IE8 inside this onload handler
-      setTimeout(function() { self.lazyLoadThumbnails(); }, 100);
-    })
-    .one('error', function() {
-      // Remove class so we no longer count as loading
-      $(this).removeClass('BRlazyloading');
-    })
-
-  //the width set with .attr is ignored by Internet Explorer, causing it to show the image at its original size
-  //but with this one line of css, even IE shows the image at the proper size
-    .css({
-      'width': $(dummyImage).width() + 'px',
-      'height': $(dummyImage).height() + 'px'
-    })
-    .attr({
-      'width': $(dummyImage).width(),
-      'height': $(dummyImage).height(),
-      'src': $(dummyImage).data('srcURL'),
-      'alt': 'Loading book image'
-    });
+/**
+ * Replaces placeholder image with real one
+ *
+ * @param {$lazyLoadImgPlaceholder} - $imgPlaceholder
+ */
+BookReader.prototype.lazyLoadImage = function ($imgPlaceholder) {
+  const leaf =  $($imgPlaceholder).data('leaf');
+  const reduce =  $($imgPlaceholder).data('reduce');
```

Comment:
No `$` since this doesn't look like a jquery object.

```suggestion
 * @param {HTMLImageElement} - imgPlaceholder
 */
BookReader.prototype.lazyLoadImage = function (imgPlaceholder) {
  const leaf =  $(imgPlaceholder).data('leaf');
  const reduce =  $(imgPlaceholder).data('reduce');
```